app/updater.js

```diff
@@ -1,32 +1,28 @@
 const { app, BrowserWindow, dialog } = require('electron')
 const os = require('os')
-const autoUpdater = require('electron-updater').autoUpdater
+const isDev = require('electron-is-dev')
+const { autoUpdater } = require('electron-updater')
 
 exports.init = () => {
   const platform = os.platform()
-  if (platform === 'linux') {
+  if (isDev || platform === 'linux') {
     return
   }
 
   autoUpdater.signals.updateDownloaded(versionInfo => {
     const dialogOptions = {
       type: 'question',
       defaultId: 0,
-      message: `The update is ready to install, Version ${versionInfo.version} has been downloaded and will be automatically installed`
+      message: `The update for version ${versionInfo.version} is ready to install, do you want to restart the app now?`,
+      buttons: ['OK', 'Cancel']
     }
 
-    const focusedWindow = BrowserWindow.getFocusedWindow()
+    const [win] = BrowserWindow.getAllWindows()
 
-    BrowserWindow.getAllWindows()
-
-    dialog.showMessageBox(focusedWindow, dialogOptions, () => {
-      setImmediate(() => {
-        app.removeAllListeners('window-all-closed')
-        if (focusedWindow !== null) {
-          focusedWindow.close()
-        }
```

Comment:
This is because Electron recommends that you handle the `window-all-closed` even, which is fired when all windows are closed but the process is still running.

In Mac you shouldn't really close the app unless the user presses `Cmd+Q` so this event allows you to handle that case.

This function disables all those event listeners so that the app can really finish executing after all windows have been closed.